src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -992,12 +979,47 @@ private void UpdateWhenChainDownloaded(IEnumerable<FederationWallet> wallets, Da
                 TimeSpans.FiveSeconds);
         }
 
+        /// <inheritdoc />
+        public void EnableFederation(string password)
+        {
+            Guard.NotEmpty(password, nameof(password));
+
+            // Protect against de-activation if the federation is already active.
+            if (this.isFederationActive)
+            {
+                this.logger.LogTrace("(-):[FEDERATION_ALREADY_ACTIVE]");
+                return;
+            }
+
+            try
+            {
+                Key key = Key.Parse(this.Wallet.EncryptedSeed, password, this.Wallet.Network);
+
+                this.Secret = new WalletSecret() { WalletPassword = password };
+                this.isFederationActive = key.PubKey.ToHex() == this.federationGatewaySettings.PublicKey;
+
+                if (!this.isFederationActive)
+                    this.logger.LogInformation("The wallet public key {0} does not match the federation member's public key {1}", key.PubKey.ToHex(), this.federationGatewaySettings.PublicKey);
+            }
+            catch (Exception ex)
+            {
+                throw new SecurityException(ex.Message);
+            }
+        }
+
         /// <inheritdoc />
         public void ImportMemberKey(string password, string mnemonic, string passphrase)
         {
             Guard.NotEmpty(password, nameof(password));
             Guard.NotEmpty(mnemonic, nameof(mnemonic));
 
+            // Protect against de-activation if the federation is already active.
+            if (this.isFederationActive)
+            {
+                this.logger.LogTrace("(-):[FEDERATION_ALREADY_ACTIVE]");
```

Comment:
`this.logger.LogWarning` here too please.

Reply:
Done.